core/src/main/scala/org/apache/spark/ui/UIUtils.scala

```diff
@@ -143,12 +143,12 @@ private[spark] object UIUtils extends Logging {
 
   // Yarn has to go through a proxy so the base uri is provided and has to be on all links
   def uiRoot: String = {
-    if (System.getenv("APPLICATION_WEB_PROXY_BASE") != null) {
-      System.getenv("APPLICATION_WEB_PROXY_BASE")
-    } else if (System.getProperty("spark.ui.proxyBase") != null) {
+    // SPARK-11484 - Use the proxyBase set by the AM, if not found then use env.
+    if (System.getProperty("spark.ui.proxyBase") != null) {
```

Comment:
I'd use the opportunity to clean this up a bit:

    sys.props("spark.ui.proxyBase").orElse(sys.env("APPLICATION_WEB_PROXY_BASE")).getOrElse("")